### PR TITLE
feat: add optimistic update for away status toggling

### LINKED ISSUE #7749: [CAL-1266] Improve UX on away toggle
[cad99e55-b438-4e8a-935a-a43a3c92e3d9](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/eace4853-6ea0-43f5-b55d-e9c9a6ba63dc/cad99e55-b438-4e8a-935a-a43a3c92e3d9)

Currently there is a massive delay as the request is sent on click then updated on the trpc refetch. This should be handled by local state then an error toast if it fails for some reason

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1266](https://linear.app/calcom/issue/CAL-1266/improve-ux-on-away-toggle)</sub>

### PR DESCRIPTION
**Pull request by O4epegb**
## What does this PR do?

Adds optimistic update for away status toggling

Fixes #7749 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Just open dropdown user menu and toggle your away status. It should be toggled instantly/optimistically. If there is an error then the toast will be shown.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

/claim #7749

### CODE REVIEW

File: packages/features/shell/Shell.tsx [hunk 1]
```diff
@@ -312,6 +312,7 @@ function UserDropdown({ small }: UserDropdownProps) {
   const { t } = useLocale();
   const { data: user } = useMeQuery();
   const { data: avatar } = useAvatarQuery();
+  const utils = trpc.useContext();
```
Comment (O4epegb) [inline packages/features/shell/Shell.tsx:315]: Just moved it up top to clearly see where it comes from before using

File: packages/features/shell/Shell.tsx [hunk 2]
```diff
@@ -324,16 +325,31 @@ function UserDropdown({ small }: UserDropdownProps) {
...
+      showToast(t("toggle_away_error"), "error");
+    },
     onSettled() {
       utils.viewer.me.invalidate();
     },
   });
-  const utils = trpc.useContext();
   const [helpOpen, setHelpOpen] = useState(false);
   const [menuOpen, setMenuOpen] = useState(false);
-  if (!user) {
```
Comment (O4epegb) [inline packages/features/shell/Shell.tsx:334]: Duplicated check, there is already another one later on

File: packages/features/shell/Shell.tsx [hunk 3]
```diff
@@ -425,16 +442,15 @@ function UserDropdown({ small }: UserDropdownProps) {
                       <Moon className={classNames("text-default", props.className)} aria-hidden="true" />
                     )}
                     onClick={() => {
-                      mutation.mutate({ away: !user?.away });
-                      utils.viewer.me.invalidate();
```
Comment (O4epegb) [inline packages/features/shell/Shell.tsx:429]: Invalidation happens inside of mutation now

**Review by sean-brydon — APPROVED** [review]
Works as expected! Nice work.

**Review by Udit-takkar — COMMENTED** [review]
@O4epegb some E2E tests are failing

**Comment by sean-brydon** [comment]
> @O4epegb some E2E tests are failing

Missed this sorry!

**Pull request merged**